march_gain_scheduling/src/march_gain_scheduling/dynamic_pid_reconfigurer.py

```diff
@@ -7,63 +7,60 @@
 
 
 class DynamicPIDReconfigurer:
-    def __init__(self, joint_list=None, max_time_step=0.1):
+    def __init__(self, joint_list):
         self._gait_type = None
         self._joint_list = joint_list
-        self._max_time_step = max_time_step
         self.interpolation_done = True
-        self.last_update_time = None
-        self._clients = []
+        self._last_update_times = []
+        self._clients = [Client('/march/controller/trajectory/gains/' + j, timeout=30) for j in self._joint_list]
         self.current_gains = []
-        for i in range(len(self._joint_list)):
-            self._clients.append(Client('/march/controller/trajectory/gains/' + self._joint_list[i], timeout=30))
         rospy.Subscriber('/march/gait/schedule/goal', GaitActionGoal, callback=self.gait_selection_callback)
         self._linearize = rospy.get_param('~linearize_gain_scheduling')
+        self._gradient = rospy.get_param('~linear_slope')
 
     def gait_selection_callback(self, data):
-        rospy.logdebug('This is the gait type: %s', data.goal.current_subgait.gait_type)
         new_gait_type = data.goal.current_subgait.gait_type
         if new_gait_type is None or new_gait_type == '':
             new_gait_type = 'walk_like'
             rospy.logwarn('The gait has no gait type, default is set to walk_like')
 
         if self._gait_type != new_gait_type or not self.interpolation_done:
-            rospy.logdebug('The selected gait: {0} is not the same as the previous gait: {1}'.format(
-                new_gait_type, self._gait_type))
-
-            self.load_current_gains()
             self._gait_type = new_gait_type
+            self.load_current_gains()
             self.interpolation_done = False
+            self._last_update_times = len(self._joint_list) * [rospy.get_time()]
```

Comment:
move to behind line 37 and input begin_time directly? If the get_param in lookup_table takes very long, this might not result in ideal behavior.
```suggestion
            self._last_update_times = len(self._joint_list) * [begin_time]
```